subsys/net/ip/connection.c

```diff
@@ -652,17 +670,21 @@ enum net_verdict net_conn_input(struct net_pkt *pkt,
 
 void net_conn_foreach(net_conn_foreach_cb_t cb, void *user_data)
 {
-	int i;
-
-	for (i = 0; i < CONFIG_NET_MAX_CONN; i++) {
-		if (!(conns[i].flags & NET_CONN_IN_USE)) {
-			continue;
-		}
+	struct net_conn *conn;
 
-		cb(&conns[i], user_data);
+	SYS_SLIST_FOR_EACH_CONTAINER(&conn_used, conn, node) {
+		cb(conn, user_data);
 	}
 }
 
 void net_conn_init(void)
 {
+	int i;
+
+	sys_slist_init(&conn_unused);
+	sys_slist_init(&conn_used);
+
+	for (i = 0; i < CONFIG_NET_MAX_CONN; i++) {
+		sys_slist_prepend(&conn_unused, &conns[i].node);
```

Comment:
Could we call here `conn_set_unused()` instead?

Reply:
it does a memset there and it would also then try to do a remove from conn_used. A bit too heavy